packages/atomic/src/components/search/result-template-components/result-template-decorators.spec.tsx

```diff
@@ -65,8 +68,8 @@ describe('resultContext method', () => {
       template: () => (
         <atomic-result
           content={document.createElement('div')}
-          engine={TestUtils.buildMockSearchAppEngine()}
           result={TestUtils.buildMockResult()}
+          interactiveResult={null as unknown as InteractiveResult}
```

Comment:
It's a spec file so no, cannot be breaking.
I'm more intrigued to why it worked before.
My guess is that Stencil 4 fixed a bug and exposed the mandatory aspect of the key.

Reply:
There is no breaking change indeed. As you said Louis, it looks like a bug.

To answer @dmbrooke  question below: 
It's... "kind of" an internal component created by the result list to do it's job when it loops over results to display.
 
I'm using the keyword "kind of" because there's no such thing as private/internal web component at the end of the day, everything is available globally in the DOM. 
So, long story short, implementers "would not actually do that in real life". 

@louis-bompart, to reduce confusion with the null/unknown cast, we could do this: 

```typescript
const mockEngine = TestUtils.buildMockSearchAppEngine();
const mockResult = TestUtils.buildMockResult();

const page = await newSpecPage({
  components: [AtomicResult],
  template: () => (
    <atomic-result
      content={document.createElement('div')}
      engine={mockEngine}
      result={mockResult}
      store={createAtomicStore()}
      interactiveResult={buildInteractiveResult(mockEngine, {
        options: {result: mockResult},
      })}
    ></atomic-result>
  ),
});

```

Reply:
Applied suggestion ✅